### PR TITLE
migrate to weakdeps

### PR DESCRIPTION
**Pull request by sjkelly**
This should save some loading and precompilation time.

### CODE REVIEW

File: ext/DataFramesExt.jl [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module DataFramesExt
+
+using PlotlyJS
+isdefined(Base, :get_extension) ? (using DataFrames) : (using ..DataFrames)
+isdefined(Base, :get_extension) ? (using CSV) : (using ..CSV)
+
+
+dataset(::Type{DataFrames.DataFrame}, name::String) = DataFrames.DataFrame(PlotlyJS.dataset(CSV.File, name))
```
Comment (thchr) [inline ext/DataFramesExt.jl:8]: Should extend the parent module's `dataset`, as in `ext/CSVExt.jl`, I think:
```suggestion
PlotlyJS.dataset(::Type{DataFrames.DataFrame}, name::String) = DataFrames.DataFrame(PlotlyJS.dataset(CSV.File, name))
```

Comment (sjkelly) [reply ext/DataFramesExt.jl:8]: Good catch, thanks!

**Comment by sjkelly** [comment]
@BeastyBlacksmith can you take a look at this please?

File: Project.toml [hunk 2]
```diff
@@ -20,16 +20,32 @@ WebIO = "0f1e0344-ec1d-5b48-a673-e5cf874b6c29"
 
 [compat]
 Blink = "0.12"
+CSV = "0.10"
+DataFrames = "1"
+IJulia = "1"
 JSExpr = "0.5, 1"
 JSON = "0.20, 0.21"
+JSON3 = "1"
```
Comment (BeastyBlacksmith) [inline Project.toml:28]: I didn't check, if these are the correct lower bounds

**Review by BeastyBlacksmith — COMMENTED** [review]
Would be good to have tests for this

**Pull request merged**